files/demo.hledger

```diff
@@ -0,0 +1,178 @@
+D USD 1,000.00  ; Format - D Currency amountFormat
+
+
+# Now let's declare our main accounts. The order they appear here is the other they will
+# show on the reports. We need to tell hledger what is the name of each account using the
+# following code: A=Asset, L=Liabilities, R=Revenues, X=Expenses, E=Equity.
+# We will use the default name, but you can change them if you like
+
+# Attention! Here and in some other places you may need at least 2 spaces, instead of 1
+
+account Asset                  A  ; Format - account name  Code  <- 2 spaces at least between name and code
+account Liabilities            L
+account Revenue                R
+account Expenses               X
+account Equity                 E
+
+
+# Let's declare some other accounts, so they will appear in the desired order
+# You can organize them in a tree structure using ':' to indicate a a parent-child relation account
+
+account Asset:Cash
+account Asset:Investments
+account Asset:Real Estate
+account Liabilities:Credit Card
+account Liabilities:Mortgage
+
+
+# Possibly you will not start from scratch. You will want to import your data from a previous software that you
+# want to adapt for the powerful hierarchical account system hledger provides. For that we will use alias so that
+# whenever hledger sees an old account, it will substitute for its new name
+
+alias jpm=Asset:Bank:JP Morgan Chase                     ; Format - alias OLD=NEW
+alias visa=Liabilities:Credit Card:Visa Platinum         ; Substitute 'visa' for 'Liabilities:Credit Card:Visa Platinum'
+
+
+# Now we will finally start adding transactions to our ledger. For that, you will need to understand the format. Don't
+# worry if it seems complicated. You will quickly be used to it, and hledger offers some great tools to make that
+# process easier with an interactive filling app with auto-complete and tools to automatically import data from your
+# bank or any other source. It may even automatically fill the data for you if it finds a similar older transaction!
+# If You you are curious, google hledger-iadd or hledger-autosync for some examples
+
+# A transaction is a financial event that has multiple posting showing where the money came and where it goes to.
+# The main property of a transaction is that it always need to balance, or in other words, it values should sum 0.
+# This property is called 'double-entry accounting'. It may seem hard to use, but hledger uses some ingenious approach
+# that makes it almost as simple as a regular 'single-entry-accounting' that you may be used to do.
+
+# Our first transaction will be the initial balances each account has one day before we effectively register transaction.
+# A good practice is to have one file for each year. Let's do it for 2018, so the date will be 31th, December 2017.
+
+2017/12/31 Initial Balance                               ; Format - date description. Date format - YYYY/MM/DD
```

Comment:
See above. I would find this example a bit weird, I recommend closing on 12/31 and opening on 1/1. Eg because I don't want to see 2017q4 showing up in my 2018 quarterly reports.

Reply:
fixing